File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n, 1);
        int i=1, prefix=1;
        while(i<n){
            prefix *= nums[i-1];
            res[i]=prefix;
            i+=1;
        }
        
        int j=n-2;
        prefix=1;
        while(j>=0){
            prefix *= nums[j+1];
            res[j] *= prefix;
            j--;
        }
        
        return res;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    let temp = [];
    temp.push(1);
    let prefix=1;
    
    for(let i=1;i<nums.length;i++){
        prefix = nums[i-1]*prefix;
        temp.push(prefix);
    }
    
    console.log(temp);
    let ps=1;
    for(let i=nums.length-2;i>=0;i--){
        ps = ps*nums[i+1];
        temp[i]= temp[i]*ps;
    }
    
    
    
    
    return temp;
};